File: homework/StudentSharedPreference/app/src/main/java/com/example/studentsharedpreference/MainActivity.java
package com.example.studentsharedpreference;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import java.util.jar.Attributes;

public class MainActivity extends AppCompatActivity {
    private Button readable;
    private Button writeable;
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;

    //SharedPreferences文件名
    private static final String SharedPreferencesFileName="student";

    private static final String ID="id";
    private static final String NAME="name";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sharedPreferences=getSharedPreferences(SharedPreferencesFileName,MODE_PRIVATE);
        editor=sharedPreferences.edit();

        readable=(Button) findViewById(R.id.readable);
        writeable=(Button) findViewById(R.id.writeable);

        writeable.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                editor.putString(ID,"2019011163");
                editor.putString(NAME,"梁矗");

                editor.apply();
            }
        });

        readable.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String id=sharedPreferences.getString(ID,"");
                String name=sharedPreferences.getString(NAME,"");
                Toast.makeText(MainActivity.this, "学号："+id+"  姓名："+name, Toast.LENGTH_SHORT).show();
            }
        });

    }
}

File: homework/RecyclerViewDemo/app/src/main/java/cn/edu/bistu/sc/se/android/recyclerviewdemo/MainActivity.kt
package cn.edu.bistu.sc.se.android.recyclerviewdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

class MainActivity : AppCompatActivity() {

    private lateinit var mRecyclerView: RecyclerView
    private var mUserWordMessageAdapter: UserMessageAdapter? = null
    private lateinit var mEditTextUserWordMessage: EditText
    private var mUserWordMessageList = mutableListOf<UserMessage>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mEditTextUserWordMessage = findViewById(R.id.edittextUserMessage)


        mRecyclerView = findViewById(R.id.recyclerview)
        mUserWordMessageAdapter = UserMessageAdapter(mUserWordMessageList)
        val linearLayoutManager = LinearLayoutManager(this)
        mRecyclerView.setLayoutManager(linearLayoutManager)
        mRecyclerView.setAdapter(mUserWordMessageAdapter)
    }

    /**
     * 用户按下消息发送按钮时发送消息
     * @param view
     */
    fun onButtonSendMessageClick(view: View?) {
        var content = mEditTextUserWordMessage.text.toString()
        content = content.trim { it <= ' ' }
        if (content.length == 0) {
            Toast.makeText(this, "文本内容不能为空", Toast.LENGTH_SHORT).show()
            return
        }
        addSendMessage(content)
        /**
         * 这里模拟接收对方发送的消息，在实际程序中，应该定时从网络获得相关数据
         */
        addReceiveMessage("我接收到$content")
    }

    /**
     * 增加接收消息
     * @param content
     */
    private fun addReceiveMessage(content: String) {
        val userWordMessage = UserMessage(content, TYPE_RECEIVE)
        mUserWordMessageList.add(userWordMessage)

        // 表示在消息的末尾插入内容
        mUserWordMessageAdapter!!.notifyItemInserted(mUserWordMessageList!!.size - 1)

        // 让 RecyclerView 自动滚动到最底部
        mRecyclerView!!.scrollToPosition(mUserWordMessageList!!.size - 1)
        mEditTextUserWordMessage!!.setText("")
    }

    /**
     * 增加发送消息
     * @param content
     */
    private fun addSendMessage(content: String) {
        val userWordMessage = UserMessage(content, TYPE_SEND)
        mUserWordMessageList.add(userWordMessage)

        // 表示在消息的末尾插入内容
        mUserWordMessageAdapter!!.notifyItemInserted(mUserWordMessageList!!.size - 1)

        // 让 RecyclerView 自动滚动到最底部
        mRecyclerView!!.scrollToPosition(mUserWordMessageList!!.size - 1)
        mEditTextUserWordMessage!!.setText("")
    }
}